### ISSUE TITLE
minprotobuf limits message size to 4mb

### ISSUE THREAD

**Issue by dryajov** [status-im/nim-codex#308]
TL;DR

- minprotobuf is limited to 4mb message sizes
- manifests can hold up to ~131072 sha256  hashes
- largest file size that we can represent with a 4mb manifest is roughly ~8.5GB
- we can fix this with nested manifests
- it would be nice to set a more reasonable limit on manifest sizes - say ~10MB
- this needs allowing to specify/increasing the max size of the message in minprotobuf
- this would be needed anyway when exporting a manifest - even with nested manifests, you'd need to store all the manifest files together when exporting/importing

[minprotobuf](https://github.com/status-im/nim-libp2p/blob/unstable/libp2p/protobuf/minprotobuf.nim#L23) imposes a `4mb` size limit on messages. 

There isn't a limit that the standard imposes, but it largely makes sense to keep messages around a couple of megabytes as per the protobuf authors advice - https://stackoverflow.com/questions/34128872/google-protobuf-maximum-size. However, there are some legit use cases that might require larger message sizes, in our case, the `manifest` size grows proportionally to the block size and total dataset size and it could reach sizes of several dozens of megabytes. We have ways to mitigate this with nested manifests, but this needs to be implemented on the other hand, it would be nice if minprotobuf allowed to specify the max message size.
